app/src/components/BookDetails/BookDetails.js

```diff
@@ -0,0 +1,79 @@
+import React, { Component } from 'react'
+
+
+import './BookDetails.css'
+
+export default class BookDetails extends Component {
+  constructor( props ) {
+    super( props )
+    this.state = {
+      id: this.props.id,
+      book: {}
+    }
+  }
+
+  componentDidMount() {
+    let fetchURL = `/books/${this.state.id}`
+    fetch( fetchURL )
+     .then( response => response.json() )
+     .then( json => json.data )
+     .then( book => {
+       this.setState({ book: book[0] })
+     })
+  }
+
+  handleDelete() {
+
+    //TODO: Add a modal for confirmation.
+    let id = this.state.id
+    let fetchURL = `/books/${id}`
+
+    let request = new Request( fetchURL, {
+      method: 'DELETE'
+    })
+
+    fetch( request )
+      .then( () => {
+        this.context.router.push( '/book-list' )
+      })
+  }
+
+  render() {
+    let book = this.state.book
```

Comment:
I think you can do two things here to make this code cuter:
1. you can user Object deconstruction.
2. you can use cosnt instead of let
so:
```
const { book } = this.state
```